nuxeo/uploads.py

```diff
@@ -98,7 +97,8 @@ def upload(self, batch, blob):
             raw=True,
             headers=headers
         )
-
+        if isinstance(blob, FileBlob):
```

Comment:
Could you encapsulate in a `try...finally` to prevent leaks? (As it was first, sorry :) )